util/json.c

```diff
@@ -220,7 +220,7 @@ match_str(const uint8_t * buf, const uint8_t * end, const char * s,
 	do {
 		if (buf == end)
 			return (end);
-		ch = *buf++;
+		ch = (char)(*buf++);
```

Comment:
I'm not changing the `char ch` on line 214 of this function because it's compared to `char s[0]` on line 275.

However, to make clang happy, I'm casting the value it gets from `buf` here.